File: src/jvmMain/java/ess/papyrus/VarFlt.java
package ess.papyrus;

import java.nio.ByteBuffer;

/**
 * Variable that stores a float.
 */
final public class VarFlt extends Variable {

    public VarFlt(ByteBuffer input) {
        this.VALUE = input.getFloat();
    }

    public VarFlt(float val) {
        this.VALUE = val;
    }

    public float getValue() {
        return this.VALUE;
    }

    @Override
    public int calculateSize() {
        return 5;
    }

    @Override
    public void write(ByteBuffer output) {
        this.getType().write(output);
        output.putFloat(this.VALUE);
    }

    @Override
    public VarType getType() {
        return VarType.FLOAT;
    }

    @Override
    public String toValueString() {
        //return String.format("%f", this.VALUE);
        return Float.toString(this.VALUE);
    }

    @Override
    public String toString() {
        //return String.format("%s:%f", this.getType(), this.VALUE);
        return this.getType() + ":" + this.toValueString();
    }

    final private float VALUE;
}

File: src/jvmMain/kotlin/ess/papyrus/VarFlt.kt
package ess.papyrus

import PlatformByteBuffer

/**
 * Variable that stores a float.
 */
class VarFlt : Variable {
    constructor(input: PlatformByteBuffer) {
        value = input.getFloat()
    }

    constructor(`val`: Float) {
        value = `val`
    }

    override fun calculateSize(): Int {
        return 5
    }

    override fun write(output: PlatformByteBuffer?) {
        type.write(output)
        output?.putFloat(value)
    }

    override val type: VarType
        get() = VarType.FLOAT

    override fun toValueString(): String {
        //return String.format("%f", this.VALUE);
        return value.toString()
    }

    override fun toString(): String {
        //return String.format("%s:%f", this.getType(), this.VALUE);
        return type.toString() + ":" + toValueString()
    }

    val value: Float
}